EFCore.CheckConstraints/Internal/EnumCheckConstraintConvention.cs

```diff
@@ -54,41 +55,63 @@ public virtual void ProcessModelFinalizing(IConventionModelBuilder modelBuilder,
                     continue;
                 }
 
-                var enumValues = Enum.GetValuesAsUnderlyingType(propertyType).Cast<object>().Distinct().ToArray();
-                if (enumValues.Length == 0)
+                if (!_cachedConstraints.TryGetValue(typeMapping, out var constraintSql))
                 {
-                    continue;
+                    if (!TryGenerateCheckConstraint(typeMapping, out constraintSql))
+                    {
+                        continue;
+                    }
+
+                    _cachedConstraints[typeMapping] = sql.ToString();
```

Comment:
Shouldn't this be `constraintSql`? We haven't actually added anything to the sql string builder yet.